File: backend/app/schemas/artwork.py
from pydantic import BaseModel

from app.utils.artwork import ArtworkType


class ArtworkUploadResponse(BaseModel):
    archive_entry_id: str
    artwork_type: ArtworkType
    file_path: str
    screenshot_id: str | None = None
    caption: str | None = None


class ArtworkReplaceResponse(ArtworkUploadResponse):
    pass


class ArtworkDeleteResponse(BaseModel):
    archive_entry_id: str | None = None
    artwork_type: ArtworkType | None = None
    screenshot_id: str | None = None
    deleted: bool = False


class ArtworkMissingResponse(BaseModel):
    archive_entry_id: str
    title: str
    missing_types: list[ArtworkType]


class ArtworkValidationResult(BaseModel):
    archive_entry_id: str
    title: str
    verification_status: str
    missing_types: list[ArtworkType]

File: frontend/lib/types/artwork.ts
export type ArtworkType = "cover" | "banner" | "logo" | "screenshot";

export interface ArtworkMissingItem {
  archive_entry_id: string;
  title: string;
  missing_types: ArtworkType[];
}

export interface ArtworkUploadResponse {
  archive_entry_id: string;
  artwork_type: ArtworkType;
  file_path: string;
  screenshot_id?: string | null;
  caption?: string | null;
}

export interface ArtworkReplaceResponse {
  archive_entry_id: string;
  artwork_type: ArtworkType;
  file_path: string;
  screenshot_id?: string | null;
  caption?: string | null;
}

export interface ArtworkDeleteResponse {
  archive_entry_id?: string | null;
  artwork_type?: ArtworkType | null;
  screenshot_id?: string | null;
  deleted: boolean;
}
